pkg/adaptor/server.go

```diff
@@ -0,0 +1,141 @@
+// (C) Copyright Confidential Containers Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package adaptor
+
+import (
+	"context"
+	"log"
+	"net"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/confidential-containers/cloud-api-adaptor/pkg/adaptor/cloud"
+	"github.com/confidential-containers/cloud-api-adaptor/pkg/adaptor/proxy"
+	"github.com/confidential-containers/cloud-api-adaptor/pkg/adaptor/vminfo"
+	"github.com/confidential-containers/cloud-api-adaptor/pkg/podnetwork"
+
+	"github.com/containerd/ttrpc"
+	pbHypervisor "github.com/kata-containers/kata-containers/src/runtime/protocols/hypervisor"
+
+	pbPodVMInfo "github.com/confidential-containers/cloud-api-adaptor/proto/podvminfo"
+)
+
+var logger = log.New(log.Writer(), "[adaptor] ", log.LstdFlags|log.Lmsgprefix)
+
+const (
+	DefaultSocketPath = "/run/peerpod/hypervisor.sock"
+	DefaultPodsDir    = "/run/peerpod/pods"
+	DefaultPauseImage = "k8s.gcr.io/pause:3.7"
+)
+
+type ServerConfig struct {
+	SocketPath    string
+	CriSocketPath string
+	PauseImage    string
+	PodsDir       string
+	ForwarderPort string
+}
+
+type Server interface {
+	Start(ctx context.Context) error
+	Shutdown() error
+	Ready() chan struct{}
+}
+
+type server struct {
+	socketPath string
+
+	ttRpc         *ttrpc.Server
+	cloudService  cloud.Service
+	vmInfoService pbPodVMInfo.PodVMInfoService
+
+	workerNode podnetwork.WorkerNode
+
+	readyCh  chan struct{}
+	stopCh   chan struct{}
+	stopOnce sync.Once
+}
+
+func NewServer(provider cloud.Provider, cfg *ServerConfig, workerNode podnetwork.WorkerNode) Server {
+
+	logger.Printf("server config: %#v", cfg)
+
+	agentFactory := proxy.NewFactory(cfg.PauseImage, cfg.CriSocketPath)
+	cloudService := cloud.NewService(provider, agentFactory, workerNode, cfg.PodsDir, cfg.ForwarderPort)
+	vmInfoService := vminfo.NewService(cloudService)
+
+	s := &server{
+		socketPath:    cfg.SocketPath,
+		cloudService:  cloudService,
+		vmInfoService: vmInfoService,
+		workerNode:    workerNode,
+		readyCh:       make(chan struct{}),
+		stopCh:        make(chan struct{}),
+	}
+
+	return s
```

Comment:
Since `s` is not been used furthermore. How about just?

```suggestion
return &server{
		socketPath:    cfg.SocketPath,
		cloudService:  cloudService,
		vmInfoService: vmInfoService,
		workerNode:    workerNode,
		readyCh:       make(chan struct{}),
		stopCh:        make(chan struct{}),
	}
```

Reply:
done